### ISSUE TITLE
multi-line Format strings get wrapped

### ISSUE THREAD

**Issue by raphael-proust** [ocaml-ppx/ocamlformat#868]
We use multi-line format “strings” (of the `format` type) like those:

```
$ cat format_strings.ml
let () =
  if true then (* Shrinking the margin a bit *)
    Format.printf
      "@[<v 2>@{<warning>@{<title>Warning@}@}@,@,\
      \        These are @{<warning>NOT@} the Droids you are looking for!@,\
       @,\
      \        Some more text. Just more letters and words.@,\
      \        All this text is left-aligned because it's part of the UI.@,\
      \        It'll be easier for the user to read this message.@]@\n@."
```
The identation and styling of the `format` string makes it obvious that the text is correctly indented. It mimics the output:

```
$ ocaml format_strings.ml
Warning

          These are NOT the Droids you are looking for!

          Some more text. Just more letters and words.
          All this text is left-aligned because it's part of the UI.
          It'll be easier for the user to read this message.
```

`ocamlformat` melts and wraps these multi-line format strings:

```
$ ocamlformat format_strings.ml
let () =
  if true then
    (* Shrinking the margin a bit *)
    Format.printf
      "@[<v 2>@{<warning>@{<title>Warning@}@}@,@,        These are \
       @{<warning>NOT@} the Droids you are looking for!@,@,        Some more \
       text. Just more letters and words.@,        All this text is \
       left-aligned because it's part of the UI.@,        It'll be easier for \
       the user to read this message.@]@\n\
       @."
```

----------------------------

For info:
```
$ ocamlformat --version
0.9
```

**Comment by gpetiot** [comment]
What value of 'wrap_string_literals' are you using? It looks like you are using 'wrap' (by default), maybe you should try another value.

**Comment by raphael-proust** [comment]
I used the default (no `.ocamlformat`) configuration. I considered the multi-line format wrapping to be an issue because with the same setup, I get much more pleasant results on strings:

```
$ cat multiline_strings.ml
let s =
  "Warning\n\
  \        These are NOT the Droids you are looking for!\n\
  \n\
  \        Some more text. Just more letters and words.\n\
  \        All this text is left-aligned because it's part of the UI.\n\
  \        It'll be easier for the user to read this message.\n"
;;
print_endline s
$ ocaml multiline_strings.ml
Warning
        These are NOT the Droids you are looking for!

        Some more text. Just more letters and words.
        All this text is left-aligned because it's part of the UI.
        It'll be easier for the user to read this message.

$ ocamlformat multiline_strings.ml
let s =
  "Warning\n\
  \        These are NOT the Droids you are looking for!\n\n\
  \        Some more text. Just more letters and words.\n\
  \        All this text is left-aligned because it's part of the UI.\n\
  \        It'll be easier for the user to read this message.\n"

;;
print_endline s
```

Is there a reason strings and format-strings are treated differently?

**Comment by gpetiot** [comment]
Fixed with #887
